src/Form/QuestionType/QuestionTypeDateAndTime.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Form\QuestionType;
+
+use Glpi\Application\View\TemplateRenderer;
+use Glpi\Form\Question;
+use Override;
+
+/**
+ * "Date and time" questions represent date picker inputs.
+ */
+abstract class QuestionTypeDateAndTime implements QuestionTypeInterface
```

Comment:
1. It would be better to have an `Abstract` prefix to the class name.
2. Having `QuestionType` prefix for each method is a bit repetitive. Maybe `XxxType` instead of `QuestionTypeXxx` would be more appropriate. @AdrienClairembault 

```suggestion
abstract class AbstractQuestionTypeDateAndTime implements QuestionTypeInterface
```